src/app/static/src/app/store/downloadResults/actions.ts

```diff
@@ -12,24 +12,36 @@ export interface DownloadResultsActions {
     prepareSpectrumOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
     prepareCoarseOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
     prepareComparisonOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
+    prepareAgywTool: (store: ActionContext<DownloadResultsState, RootState>) => void
     prepareOutputs: (store: ActionContext<DownloadResultsState, RootState>) => void
     poll: (store: ActionContext<DownloadResultsState, RootState>, downloadType: string) => void
     downloadComparisonReport: (store: ActionContext<DownloadResultsState, RootState>) => void
     downloadSpectrumOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
     downloadSummaryReport: (store: ActionContext<DownloadResultsState, RootState>) => void
     downloadCoarseOutput: (store: ActionContext<DownloadResultsState, RootState>) => void
+    downloadAgywTool: (store: ActionContext<DownloadResultsState, RootState>) => void
 }
 
 export const actions: ActionTree<DownloadResultsState, RootState> & DownloadResultsActions = {
 
     async prepareOutputs(context) {
         const {dispatch} = context
-        await Promise.all([
-            dispatch("prepareCoarseOutput"),
-            dispatch("prepareSummaryReport"),
-            dispatch("prepareSpectrumOutput"),
-            dispatch("prepareComparisonOutput")
-        ]);
+        if (switches.agywDownload) {
+            await Promise.all([
+                dispatch("prepareCoarseOutput"),
+                dispatch("prepareSummaryReport"),
+                dispatch("prepareSpectrumOutput"),
+                dispatch("prepareComparisonOutput"),
+                dispatch("prepareAgywTool"),
+            ]);
+        } else {
+            await Promise.all([
+                dispatch("prepareCoarseOutput"),
+                dispatch("prepareSummaryReport"),
+                dispatch("prepareSpectrumOutput"),
+                dispatch("prepareComparisonOutput"),
+            ]);
+        }
```

Comment:
yep so you can conditionally add this last element to an array by using a ternary operator like so:
```typescript
const promisesArray = [
    dispatch("prepareCoarseOutput"),
    dispatch("prepareSummaryReport"),
    dispatch("prepareSpectrumOutput"),
    dispatch("prepareComparisonOutput"),
    ...switches.agywDownload ? [dispatch("prepareAgywTool")] : []
]
await Promise.all(promisesArray)
```
this will unpack the arrays in the ternary and not add anything if you spread an empty array

Reply:
Thanks, that is looking better. I tried this before but was getting IntelliJ warnings about missing await for an async function call but I guess it is fine in this case as IntelliJ just not picking up on the fact that we are awaiting that later in the call

Reply:
vscode >> intellij :yum: